GVFS/GVFS.Platform.Windows/ETWTelemetryEventListener.cs

```diff
@@ -85,62 +83,60 @@ public override void Dispose()
             EventSourceOptions options = this.CreateOptions(level, keywords, opcode);
             EventSource.SetCurrentThreadActivityId(activityId);
 
-            if (jsonPayload != null)
+            if (string.IsNullOrEmpty(jsonPayload))
             {
-                JsonPayload payload = new JsonPayload(jsonPayload, this.enlistmentId, this.mountId);
+                jsonPayload = "{}";
+            }
+
+            if (string.IsNullOrEmpty(this.ikey))
+            {
+                var payload = new
+                {
+                    Json = jsonPayload,
+                    EnlistmentId = this.enlistmentId,
+                    MountId = this.mountId,
+                };
                 this.eventSource.Write(eventName, ref options, ref activityId, ref parentActivityId, ref payload);
             }
             else
             {
-                Payload payload = new Payload(this.enlistmentId, this.mountId);
+                var payload = new
+                {
+                    PartA_iKey = this.ikey,
+                    Json = jsonPayload,
+                    EnlistmentId = this.enlistmentId,
+                    MountId = this.mountId,
+                };
                 this.eventSource.Write(eventName, ref options, ref activityId, ref parentActivityId, ref payload);

```

Comment:
The call to `this.eventSource.Write` is the same in both the `if` and the `else` block, can it be moved out?

Reply:
Because it is using an anonymous type the `Write` is not using the same type for the last parameter.